feat(datepicker): add support for disabled datepicker

jnizet opened the pull request:
fix #702

pkozlowski-opensource commented:
Wooohooo, thnx for this, it was blocking me for another PR :-) Will review tomorrow when fresh.

pkozlowski-opensource commented:
@maxokorokov would you mind reviewing this one?

pkozlowski-opensource commented on demo/src/app/components/datepicker/demos/basic/datepicker-basic.ts:
```diff
@@ -9,8 +9,13 @@ const now = new Date();
     this.model = {year: now.getFullYear(), month: now.getMonth(), day: now.getDate()};
   }
+
+  toggleDisabled() {
```
We could probably in-line this in a template. Having said this I'm not sure it makes sense to demonstrate it as part of the very basic demo. If I recall properly we want to have to the basic demo as simple as possible (that is, use default values for all props and provide only ones that are must-have for the most rudimentary demo).

pkozlowski-opensource commented on src/datepicker/datepicker.ts:
```diff
@@ -41,6 +41,7 @@ const NGB_DATEPICKER_VALUE_ACCESSOR = {
         [dayTemplate]="dayTemplate || dt"
         [showWeekdays]="showWeekdays"
         [showWeekNumbers]="showWeekNumbers"
+        [disabled]="disabled"
```
shouldn't we disable navigation as well?

jnizet replied:
I wasn't sure, so I chose the easy route :-)

But yes, that's another thing we have to decide. It isn't strictly necessary, but it might make things clearer. If we choose to do it, I guess it would also be a good idea to switch to the month of the selected date when it becomes disabled.

jnizet commented:
Rebased and added a specific demo for the disabled state.
Added a second commit to also disable the navigation. Not sure yet if it's a good idea or not, but since it's in a separate commit, it'll be easy to pick one or both depending on what we choose.

maxokorokov commented:
LGTM. And I think there is no point in leaving navigation enabled. But no strong opinions on this one, as I guess that the datepicker will rarely be used inline. And in the case of an input + datepicker in the popup its the input that should be disabled, right?

maxokorokov commented on src/datepicker/datepicker-month-view.spec.ts:
```diff
@@ -107,6 +107,22 @@ describe('ngbDatepickerMonthView', () => {
+    const dates = getDates(fixture.nativeElement);
+    dates[0].click();
+
+    expect(fixture.componentInstance.onClick).not.toHaveBeenCalledWith();
```
better be `not.toHaveBeenCalled()` but I guess it's my mistake from the test case above (line 107) :)

jnizet commented:
Replaced `toHaveBeenCalledWith()` by `toHaveBeenCalled()`.
